pkg/orchestration/wiring/kubeingress/plugin.go

```diff
@@ -201,7 +199,42 @@ func buildIngressResource(serviceName smith_v1.ResourceName, resource *orch_v1.S
 		Exposed: true,
 	}
 
-	return ingressResource, nil
+	return &ingressResource, nil
+}
+
+// buildIngressResource constructs the Kube / KITT Ingress object
+// with a default rule, plus all alias rules from dependant internalDNS resources
+func buildIngressResource(serviceName smith_v1.ResourceName, resource *orch_v1.StateResource, context *wiringplugin.WiringContext) (*wiringplugin.WiredSmithResource, error) {
+	var timeout = defaultContourTimeout
+
+	if resource.Defaults != nil {
+		var rawDefaultsSpec Spec
+		if err := json.Unmarshal(resource.Defaults.Raw, &rawDefaultsSpec); err != nil {
+			return nil, errors.WithStack(err)
+		}
+		if rawDefaultsSpec.IngressTimeout != nil {
+			timeout = *rawDefaultsSpec.IngressTimeout
+		}
+	}
+
+	if resource.Spec != nil {
+		var rawIngressSpec Spec
+		if err := json.Unmarshal(resource.Spec.Raw, &rawIngressSpec); err != nil {
+			return nil, errors.WithStack(err)
+		}
+
+		if rawIngressSpec.IngressTimeout != nil {
+			timeout = *rawIngressSpec.IngressTimeout
+
+			if timeout < 1 || timeout > 300 {
```

Comment:
I don't think this type of validation belongs here but I didn't see where to put it.

Reply:
Validation story is not great ATM, we need to invest more into it. It's on my plate, but too busy atm.